src/ui/widgets/markereditor.cpp

```diff
@@ -80,11 +84,19 @@ void MarkerEditor::setup()
 
     connect(addButton, &QPushButton::clicked,
             this, [this]() {
-        auto item = new QTreeWidgetItem(mTree);
         const int frame = mScene ? mScene->getCurrentFrame() : 0;
+        for (int i = 0; i < mTree->topLevelItemCount(); ++i) {
+            auto existingItem = mTree->topLevelItem(i);
+            if (existingItem && existingItem->data(0, Qt::UserRole).toInt() == frame) {
+                return;
+            }
+        }
+        auto item = new QTreeWidgetItem(mTree);
+        mTree->blockSignals(true);
         item->setText(1, QString::number(frame));
         item->setText(0, QString::number(frame));
         item->setData(0, Qt::UserRole, frame);
+        mTree->blockSignals(false);
```

Comment:
This does nothing? the block is removed before anything happens in mTree.

Reply:
Remember that I commented that "if you have a marker in frame 0 and add a new one at any other frame with the "+" button of the Marker Editor, it deletes any marker created at frame 0...". My research found that when creating a new row in the Markers Editor window, it launches too the `itemChanged` function, taking the current `oframe`and `frame` values that at that point are both just initialized with value 0 and ending up with deleting the marker at frame 0.
Blocking the signals for line 96, 97 and 98 ensures there is not update in that range but later when `oframe` and `frame` really take the values of the current "time line" position and the rest works as expected.